### PR TITLE
Feat/modal on will dismiss

### PR DESCRIPTION
**Pull request by MonsterGameDev**
This PR closes # (reference issue number here)

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] cookbook application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Modal API did not provide onWillDismiss

## What is the new behavior?
Now it does ;-)

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: apps/cookbook/src/app/examples/modal-example/modal-example.component.ts [hunk 1]
```diff
@@ -83,8 +83,13 @@ this.modal?.scrollToBottom();`,
 @ViewChild('nameInput', { static: false, read: ElementRef }) private nameInputElement: ElementRef<HTMLInputElement>;
 
 ngOnInit() {
-  this.modal?.didPresent.then(() => this.nameInputElement?.nativeElement.focus());
+  this.modal?.didPresent.then(() => this.nameInputElement?.nativeElement.focus();
```
Comment (jakobe) [inline apps/cookbook/src/app/examples/modal-example/modal-example.component.ts:86]: Typo
```suggestion
  this.modal?.didPresent.then(() => this.nameInputElement?.nativeElement.focus()); 
```

File: apps/cookbook/src/app/showcase/modal-showcase/modal-showcase.component.ts [hunk 2]
```diff
@@ -67,6 +67,11 @@ export class ModalShowcaseComponent implements AfterViewInit {
       description: 'Emitted when the modal is ready and the enter animation has finished',
       inputValues: ['Promise<void>'],
     },
+    {
+      name: 'willDismiss',
```
Comment (jakobe) [inline apps/cookbook/src/app/showcase/modal-showcase/modal-showcase.component.ts:71]: I know this is the Ionic jargon, but maybe align with existing Kirby wording?
```suggestion
      name: 'willClose',
```

File: apps/cookbook/src/app/showcase/modal-showcase/modal-showcase.component.ts [hunk 3]
```diff
@@ -67,6 +67,11 @@ export class ModalShowcaseComponent implements AfterViewInit {
...
+      description: 'Emitted when the modal receives a dismiss (close) command',
```
Comment (jakobe) [inline apps/cookbook/src/app/showcase/modal-showcase/modal-showcase.component.ts:72]:
```suggestion
      description: 'Emitted when the user closes the modal or modal.close() method is called',
```

File: libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.ts [hunk 4]
```diff
@@ -73,6 +75,8 @@ export class ModalWrapperComponent implements Modal, AfterViewInit, OnInit, OnDe
   ngOnInit(): void {
     this.ionModalElement = this.elementRef.nativeElement.closest('ion-modal');
     this.listenForIonModalDidPresent();
+    this.listenForIonModalWillDismiss();
+
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.ts:79]: Whitespace :)
```suggestion
```

File: libs/designsystem/src/lib/components/modal/services/modal.interfaces.ts [hunk 5]
```diff
@@ -12,6 +12,7 @@ export interface Overlay {
 
 export abstract class Modal {
   didPresent: Promise<void>;
+  willDismiss: Promise<void>;
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/modal/services/modal.interfaces.ts:15]: I know this is the Ionic jargon, but maybe align with existing Kirby wording?
```suggestion
  willClose: Promise<void>;
```

Comment (MonsterGameDev) [reply libs/designsystem/src/lib/components/modal/services/modal.interfaces.ts:15]: done

File: apps/cookbook/src/app/showcase/modal-showcase/modal-showcase.component.ts (hunk 3 above)
Comment (MonsterGameDev) [reply apps/cookbook/src/app/showcase/modal-showcase/modal-showcase.component.ts:72]: done

File: libs/designsystem/src/lib/components/modal/services/modal.interfaces.ts (hunk 5 above)
Comment (MonsterGameDev) [reply libs/designsystem/src/lib/components/modal/services/modal.interfaces.ts:15]: done

**Pull request merged**